spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/cassandra/CassandraReactiveHealthIndicator.java

```diff
@@ -50,8 +55,30 @@ public CassandraReactiveHealthIndicator(ReactiveCassandraOperations reactiveCass
 
 	@Override
 	protected Mono<Health> doHealthCheck(Health.Builder builder) {
-		return this.reactiveCassandraOperations.getReactiveCqlOperations().queryForObject(SELECT, String.class)
-				.map((version) -> builder.up().withDetail("version", version).build()).single();
+
+		return this.reactiveCassandraOperations.getReactiveCqlOperations().execute(extractMetadata()).single()
+				.map((metadata) -> buildHealth(builder, metadata));
+	}
+
+	protected Health buildHealth(Builder builder, Metadata metadata) {
+		Collection<Node> nodes = metadata.getNodes().values();
+		boolean atLeastOneUp = nodes.stream().map(Node::getState).anyMatch((state) -> state == NodeState.UP);
+		if (atLeastOneUp) {
+			builder.up();
+		}
+		else {
+			builder.down();
+		}
+
+		// fill details with version of the first node (if the version is not null)
+		nodes.stream().map(Node::getCassandraVersion).filter(Objects::nonNull).findFirst()
+				.ifPresent((version) -> builder.withDetail("version", version));
+		return builder.build();
```

Comment:
Good suggestion, I was thinking about the same. I didn't extract it yet because it was duplicated for the current health check logic as well. So if the spring-boot team is ok with extracting this to a separate component, I am +1 on this.